### PR TITLE
Backport BF16 Support to Necessary Ops

### PR DESCRIPTION
**Pull request by ScXfjiang**
See details in this comment: https://github.com/ROCm/frameworks-internal/issues/9790#issue-2591852147

### CODE REVIEW

**Comment by i-chaochen** [comment]
Thanks! @ScXfjiang could sqush as one commit and list all added bf16 supports in this one commit message, please?

File: tensorflow/python/BUILD [hunk 1]
```diff
@@ -3759,7 +3759,7 @@ cuda_py_test(
     main = "ops/nn_fused_batchnorm_test.py",
     python_version = "PY3",
     shard_count = 24,
-    tags = ["no_rocm"],
+    tags = [],
```
Comment (i-chaochen) [inline tensorflow/python/BUILD]: we can remove this `tags=[],` as it's not used anymore.

**Review by i-chaochen — APPROVED** [review]
LGTM, thanks!

File: tensorflow/python/BUILD (hunk 1 above)
Comment (ScXfjiang) [reply tensorflow/python/BUILD]: done

**Comment by ScXfjiang** [comment]
> Thanks! @ScXfjiang could sqush as one commit and list all added bf16 supports in this one commit message, please?

done

**Review by pemeliya — APPROVED** [review]

**Pull request merged**